File: app/src/main/java/com/github/overpass/gather/screen/meeting/chat/users/list/viewholder/BaseViewHolder.java
package com.github.overpass.gather.screen.meeting.chat.users.list.viewholder;

import android.view.View;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.github.overpass.gather.screen.meeting.chat.users.list.UsersAdapter;
import com.github.overpass.gather.screen.meeting.chat.users.model.UserModel;

public abstract class BaseViewHolder extends RecyclerView.ViewHolder {

    public BaseViewHolder(@NonNull View itemView) {
        super(itemView);
    }

    public void bind(UserModel userModel, UsersAdapter.OnItemClickListener listener) {
    }
}

File: app/src/main/java/com/github/overpass/gather/screen/meeting/chat/users/list/viewholder/BaseViewHolder.kt
package com.github.overpass.gather.screen.meeting.chat.users.list.viewholder

import android.view.View
import androidx.recyclerview.widget.RecyclerView
import com.github.overpass.gather.screen.meeting.chat.users.list.UsersAdapter

abstract class BaseViewHolder<I : Item>(itemView: View) : RecyclerView.ViewHolder(itemView) {

    open fun bind(item: I, listener: UsersAdapter.OnItemClickListener) {}
}